ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/validation/BlockGossipValidator.java

```diff
@@ -136,7 +141,14 @@ public SafeFuture<InternalValidationResult> validate(final SignedBeaconBlock blo
                 return reject("Block signature is invalid");
               }
 
-              if (!receivedValidBlockInfoSet.add(new SlotAndProposer(block))) {
+              // We want to add blocks in the infoSet only when they come from gossip.
+              // When dealing with locally produced block, we only want to check them against seen
+              // block coming from gossip.
```

Comment:
As I understand it (correct me if I'm wrong), you are not adding local block to the seen set here because it will pass Validator twice and it will be actually added to the seen set in another run. I'd clarify this in the comment